eth/test/integration.js

```diff
@@ -152,12 +169,100 @@ contract("TreasureHuntCreator test", async accounts => {
     });
   });
 
+  describe("getLeaderboard", async () => {
+    it("should return the list of players and chapters", async () => {
+      let [player1, player2, player3] = accounts;
+
+      let testSolution1 = "A solution 1";
+      let solutionKey1 = await getSolutionAddress(testSolution1);
+
+      let testSolution2 = "A solution 2";
+      let solutionKey2 = await getSolutionAddress(testSolution2);
+
+      let testSolution3 = "A solution 3";
+      let solutionKey3 = await getSolutionAddress(testSolution3);
+
+      let instance = await TreasureHuntCreator.new(
+        [solutionKey1, solutionKey2, solutionKey3],
+        [quests[0], quests[1], quests[2]]
+      );
+
+      // Let the game begin!
+      let leaderboard;
+      let expectedLeaderboard;
+
+      // Util to merge an address with a chapter
+      let merge = (address, chapter) =>
+        new BN(address.replace("0x", ""), 16).shln(96).or(new BN(chapter));
+
+      // player 1 solves chapter 1
+      await instance.submit(
+        ...(await getSolutionSignature(testSolution1, player1)),
+        {
+          from: player1
+        }
+      );
+      leaderboard = await instance.getLeaderboard(0);
+      expectedLeaderboard = Array(64).fill(new BN(0));
+      expectedLeaderboard[0] = merge(player1, 1);
+      assert.deepEqual(
+        leaderboard.map(n => n.toString()),
+        expectedLeaderboard.map(n => n.toString())
+      );
+
+      // player 2 solves chapter 1
+      await instance.submit(
+        ...(await getSolutionSignature(testSolution1, player2)),
+        {
+          from: player2
+        }
+      );
+      leaderboard = await instance.getLeaderboard(0);
+      expectedLeaderboard = Array(64).fill(new BN(0));
+      expectedLeaderboard[0] = merge(player1, 1);
+      expectedLeaderboard[1] = merge(player2, 1);
+      assert.deepEqual(
+        leaderboard.map(n => n.toString()),
+        expectedLeaderboard.map(n => n.toString())
+      );
+
+      // plot twist: player 3 solves chapter 1, 2, and 3
+      await instance.submit(
+        ...(await getSolutionSignature(testSolution1, player3)),
+        {
+          from: player3
+        }
+      );
+      await instance.submit(
+        ...(await getSolutionSignature(testSolution2, player3)),
+        {
+          from: player3
+        }
+      );
+      await instance.submit(
+        ...(await getSolutionSignature(testSolution3, player3)),
+        {
+          from: player3
+        }
+      );
+      leaderboard = await instance.getLeaderboard(0);
+      expectedLeaderboard = Array(64).fill(new BN(0));
+      expectedLeaderboard[0] = merge(player1, 1);
+      expectedLeaderboard[1] = merge(player2, 1);
+      expectedLeaderboard[2] = merge(player3, 3);
+      assert.deepEqual(
+        leaderboard.map(n => n.toString()),
+        expectedLeaderboard.map(n => n.toString())
+      );
+    });
+  });
+
```

Comment:
can you also test what happens when you get the leaderboard and the is no player? (on a separate test maybe)